lib/reducers/create-otp-reducer.js

```diff
@@ -256,22 +256,37 @@ function createOtpReducer (config) {
     const activeSearch = state.searches[searchId]
     switch (action.type) {
       case 'ROUTING_REQUEST':
-        const { activeItinerary, pending } = action.payload
-        return update(state, {
-          searches: {
-            [searchId]: {
-              $set: {
-                activeItinerary,
-                activeLeg: null,
-                activeStep: null,
-                pending,
-                // FIXME: get query from action payload?
-                query: clone(state.currentQuery),
-                response: [],
-                timestamp: getTimestamp()
-              }
-            }
-          },
+        const {
+          activeItinerary,
+          pending,
+          updateSearchInReducer
+        } = action.payload
+        const searchUpdate = updateSearchInReducer
```

Comment:
Non-blocking: Might be good to throw this ternary into a method that's stored elsewhere?

Reply:
I'm not quite following. What would this look like?

Reply:
```js
function getSearchUpdate (payload) {
  const {
    activeItinerary,
    pending,
    updateSearchInReducer
  } = payload
  return updateSearchInReducer ? {
    activeItinerary: { $set: activeItinerary },
    activeLeg: { $set: null },
    activeStep: { $set: null },
    pending: { $set: pending },
    // FIXME: get query from action payload?
    query: { $set: clone(state.currentQuery) },
    // omit requests reset to make sure requests can be added to this
    // search
    timestamp: { $set: getTimestamp() }
  }
  : {
    $set: {
      activeItinerary,
      activeLeg: null,
      activeStep: null,
      pending,
      // FIXME: get query from action payload?
      query: clone(state.currentQuery),
      response: [],
      timestamp: getTimestamp()
    }
  }
}
```